### PR TITLE
Testing Enumerable Methods with RSpec

### PR DESCRIPTION
**Pull request by YuvrajM108**
In this pull request we did the following:
- Initialized RSpec in our project.
- Set up a spec file for the project.
- Create RSpec test cases for all custom enumerable methods.
- Correct all linter errors.
- Edit README file.

### CODE REVIEW

File: spec/enumerable_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
+require_relative '../enumerable_methods'
+
+describe Enumerable do
+  describe '#my_each' do
+    it 'prints every element in an array' do
+      arr = [1, 2, 3]
+      expect(arr.my_each { |x| puts x, '--' }).to eql(arr)
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_methods_spec.rb]: - [x] Also, write tests for negative case scenarios for this `#my_each` method. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) Example: there is a spec for correct usage of the method \[e.g. `it "creates object if all required attributes are passed"`\] and also there is spec for the wrong usage of a method \[e.g. `it "does not create the object if required attributes are missing"`\].

File: spec/enumerable_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  describe '#my_each_with_index' do
+    let(:hash) { {} }
+    it 'prints every element with the index of the element' do
+      expect(%w[cat dog wombat].each_with_index do |item, index|
```
Comment (enaburekhan) [inline spec/enumerable_methods_spec.rb]: - [x] Use the custom method `#my_each_with_index`. Currently, the regular ruby method is been used.
- [x] Also, write tests for the negative case scenarios. An example already highlighted above

File: spec/enumerable_methods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+             end).to eql(%w[cat dog wombat])
+    end
+  end
+
+  describe '#my_select' do
+    it 'prints every element with the index of the element' do
+      arr = [1, 2, 3, 4, 5]
+      expect(arr.select(&:even?)).to eql([2, 4])
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_methods_spec.rb]: - [x] Also, write tests for the negative case scenarios. An example already highlighted above

File: spec/enumerable_methods_spec.rb [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    it 'returns product of range of values from 5 to 10' do
+      expect((5..10).my_inject(1) { |product, n| product * n }).to eql(151_200)
+    end
+
+    it 'returns product of range of values from 5 to 10' do
+      expect(%w[cat sheep bear].inject do |memo, word|
+               memo.length > word.length ? memo : word
+             end).to eql('sheep')
+    end
+  end
```
Comment (enaburekhan) [inline spec/enumerable_methods_spec.rb]: - [x] Also, write tests for the negative case scenarios. An example already highlighted above

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi team!
Great job on this project:clap::clap:.** 

**Major highlights**

- The readme is very descriptive and professional
- The PR is also very descriptive

However, we just need to make a few changes to get the project approved:muscle:.

**To improve**

- [ ] Check my comments below and make the required changes.

Make the requested changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

File: spec/enumerable_methods_spec.rb (hunk 1 above)
Comment (YuvrajM108) [reply spec/enumerable_methods_spec.rb]: The only negative test scenario that I am seeing is when the method is called without a block. In this case, it is expected to return an enumerator. I am not able to find a way in which I can get the correct enumerator value to be placed into the parenthesis of eql. @enaburekhan

**Review by taiwo2 — APPROVED** [review]
# Status: Approved 🎉 

Hey @YuvrajM108 , Good Job for implementing what the previous Code Reviewer requested. 
Your project is Approved and prepare for next project. Good luck
